src/test/java/dev/mccue/resolve/maven/MavenCentralDownloaderTest.java

```diff
@@ -0,0 +1,30 @@
+package dev.mccue.resolve.maven;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public final class MavenCentralDownloaderTest {
+
+    @BeforeEach
+    public void setup() throws IOException {
+//        URL u = mock(URL.class);
+//        HttpURLConnection h = mock(HttpURLConnection.class);
+//        when(u.openConnection()).thenReturn(h);
+//        when(h.getResponseCode()).thenReturn(200);
+    }
```

Comment:
Writing a test where you mock the actual http operations is a bit obtuse, don't you think? As written this class is (conceptually) responsible for a few things

1. Constructing the URL to download a file
2. Initiating the Http Request to download a file
3. Storing the downloaded file in a directory

We already know that (3) is going to lead to a "cache" abstraction of some kind, so the way to test that would be to pass in a Cache implementation which is somewhat fake. (2) Can be solved if you were to either accept that this would be an integration test (actually download something from maven central in the test) or come up with a "transport" abstraction/interface.

Mockitio is...fine? Like it lets you really quickly make an instance of a class that is subclassable and makes all the methods throw exceptions. What you are doing when you do that though is a "whitebox" test. You **know** that the code is only calling `.openConnection`` and `.getResponseCode`.

The problem with this is that its not really giving you much confidence in anything and will also break if this class decides to not use URL and HttpUrlConnection (which there is reason not to)

Reply:
As discussed, testing won't be done for this file for now so will delete